pandas/core/arrays/arrow/array.py

```diff
@@ -2089,6 +2090,59 @@ def _dt_isocalendar(self):
     def _dt_is_leap_year(self):
         return type(self)(pc.is_leap_year(self._pa_array))
 
+    @property
+    def _dt_is_month_start(self):
+        return type(self)(pc.equal(pc.day(self._pa_array), 1))
+
+    @property
+    def _dt_is_month_end(self):
+        plus_one_day = pc.add(self._pa_array, pa.scalar(datetime.timedelta(days=1)))
+        return type(self)(pc.equal(pc.day(plus_one_day), 1))
+
+    @property
+    def _dt_is_year_start(self):
+        return type(self)(
+            pc.and_(
+                pc.equal(pc.month(self._pa_array), 1),
+                pc.equal(pc.day(self._pa_array), 1),
+            )
+        )
+
+    @property
+    def _dt_is_year_end(self):
+        return type(self)(
+            pc.and_(
+                pc.equal(pc.month(self._pa_array), 12),
+                pc.equal(pc.day(self._pa_array), 31),
+            )
+        )
+
+    @property
+    def _dt_is_quarter_start(self):
+        result = pc.equal(
+            pc.floor_temporal(self._pa_array, unit="quarter"),
+            pc.floor_temporal(self._pa_array, unit="day"),
+        )
+        return type(self)(result)
+
+    @property
+    def _dt_is_quarter_end(self):
+        result = pc.equal(
+            pc.ceil_temporal(self._pa_array, unit="quarter"),
+            pc.ceil_temporal(self._pa_array, unit="day"),
```

Comment:
Would this have to be:
```suggestion
            pc.floor_temporal(self._pa_array, unit="day"),
```

Reply:
Hmm, so I just noticed `pc.ceil_temporal(self._pa_array, unit="quarter")` will return the start of the new quarter

```
(Pdb) self._pa_array
<pyarrow.lib.ChunkedArray object at 0x14764cb30>
[
  [
    2023-11-30 03:00:00.000000,
    2023-01-01 03:00:00.000000,
    2023-03-31 03:00:00.000000,
    null
  ]
]
(Pdb) pc.ceil_temporal(self._pa_array, unit="quarter")
<pyarrow.lib.ChunkedArray object at 0x1478e1860>
[
  [
    2024-01-01 00:00:00.000000,
    2023-04-01 00:00:00.000000,
    2023-04-01 00:00:00.000000,
    null
  ]
]
(Pdb) pc.floor_temporal(self._pa_array, unit="day")
<pyarrow.lib.ChunkedArray object at 0x1478e1950>
[
  [
    2023-11-30 00:00:00.000000,
    2023-01-01 00:00:00.000000,
    2023-03-31 00:00:00.000000,
    null
  ]
]
```

so I would need to check if `pc.floor_temporal(self._pa_array, unit="day")` + 1 day = `pc.ceil_temporal(self._pa_array, unit="quarter")`?

Reply:
Indeed! How about:
```
return pc.equal(pc.days_between(pc.floor_temporal(array, unit="day"), pc.ceil_temporal(array, unit="quarter")), 1)
```